### PR TITLE
Updating instructions on dev docs and moving Make targets

### PR DESCRIPTION
**Pull request by gvegayon**
- Updates the instructions under `docs/source/developer_documentation.rst` reflecting where the tutorials live now.
- Moves all the `Makefile` targets under `./model/Makefile` to `./docs/source/tutorials/Makefile` (new Makefile).

### CODE REVIEW

File: docs/source/tutorials/Makefile
Comment (dylanhmorris) [inline docs/source/tutorials/Makefile:1]: Should this be its own thing or should it be part of `docs/Makefile`? As written, `docs/Makefile` seems to have commands that do nothing, and I think we should avoid recursive make where possible.

**Review by dylanhmorris — COMMENTED** [review]
One key question @gvegayon

File: docs/source/tutorials/Makefile
Comment (dylanhmorris) [reply docs/source/tutorials/Makefile:1]: @gvegayon flagging this

Comment (gvegayon) [reply docs/source/tutorials/Makefile:1]: The `docs/Makefile` was originally added with the Sphinx website. I tried to add the targets to build the qmd files, but that broke the Sphinx pipeline. The lines adding print info to `make help` are residual from that attempt. I've corrected that. The `docs/source/tutorials/Makefile` is a helper that contains targets to render and convert the qmd files to py and ipynb. Is the same file that used to live under `model/docs/`.

Comment (dylanhmorris) [reply docs/source/tutorials/Makefile:1]: In that case I suggest renaming the `.PHONY` targets from `docs_*` to `tutorials_*` for clarity.

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
@gvegayon one suggestion

**Comment by dylanhmorris** [comment]
@gvegayon looks like this needs a pre-commit fix but then should be ready for re-review?

**Review by dylanhmorris — APPROVED** [review]

**Pull request merged**